### ISSUE TITLE
WebSockets tests hang in Linux

### ISSUE THREAD

**Issue by roncain** [dotnet/wcf#463]
All WebSockets tests hang in Linux.  They eventually timeout but after a very long time.  In addition to a fix, it would be nice to have a shorter timeout in the tests to avoid the pain.

=== TEST EXECUTION SUMMARY ===
   Security.TransportSecurity.Tests  Total: 9, Errors: 0, Failed: 8, Skipped: 0, Time: 6.471s
One or more tests failed while running tests from 'Security.TransportSecurity.Tests'.  Exit code 8.
   Binding_NetHttp_WebSocketTests.RequestResponseOverWebSocketManually_Echo_RoundTrips_Guid [FAIL]
      System.TimeoutException : Not completed within the alloted time of 00:20:00
      Stack Trace:
            at WcfDuplexServiceCallback.get_CallbackGuid()
            at Binding_NetHttp_WebSocketTests.RequestResponseOverWebSocketManually_Echo_RoundTrips_Guid()
            at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
            at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
            at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   Binding_NetHttp_WebSocketTests.DuplexOverWebSocket_Echo_RoundTrips_Guid [FAIL]
      System.TimeoutException : Not completed within the alloted time of 00:20:00
      Stack Trace:
            at WcfDuplexServiceCallback.get_CallbackGuid()
            at Binding_NetHttp_WebSocketTests.DuplexOverWebSocket_Echo_RoundTrips_Guid()
            at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
            at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
            at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   Binding_NetHttp_WebSocketTests.TransportUsageAlways_WebSockets_Synchronous_Call [FAIL]
      System.TimeoutException : Not completed within the alloted time of 00:20:00
      Stack Trace:
            at WcfDuplexServiceCallback.get_CallbackGuid()
            at Binding_NetHttp_WebSocketTests.TransportUsageAlways_WebSockets_Synchronous_Call()
            at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
            at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
            at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
Finished:    Binding.NetHttp.Tests

**Comment by mconnew** [comment]
Although we are skipping these tests on Linux now, the fix will be to make the call throw an exception instead of swallowing it somewhere.

**Comment by roncain** [comment]
@mconnew -- that would eliminate the hang but only make us aware where it actually failed.  I recommend that after you add the exception throwing code, we retitle this to "WebSocket tests fail in Linux" and reassign as appropriate.   There is no other issue I am aware of tracking WS in Linux.

**Comment by zhenlan** [comment]
@roncain can you please look into a fix of this issue?

**Comment by roncain** [comment]
Update: only the following 3 WS tests timeout.  All others say "Web sockets is not supported on this platform".
```
WebSocketTests.WebSocket_Http_WSTransportUsageAlways_DuplexCallback_GuidRoundtrip
WebSocketTests.WebSocket_WSScheme_WSTransportUsageAlways_DuplexCallback_GuidRoundtrip
WebSocketTests.WebSocket_Http_WSTransportUsageDefault_DuplexCallback_GuidRoundtrip
```
All 3 fail with the error:
"System.TimeoutException : Not completed within the alloted time of 00:01:00:

**Comment by roncain** [comment]
Now that this issue is related to only 3 tests, and they all "successfully" timeout after 1 minute, does it make sense to continue investigating this before we have fixed #420 assigned to @mconnew?  I tried enabling all WS tests, and on Ubuntu they all say "WS not supported on this platform" except for the 3 mentioned in this current issue.

/cc @zhenlan @mconnew

**Comment by zhenlan** [comment]
Since this is not as bad as before (hang for 20 minutes) anymore, I think it's okay to not do anything with this issue and wait Matt to validate that the WebSocket on Linux is fixed.
